lblod/ember-rdfa-editor
# Move property/backlink calculation/annotation out of say-parser

### Overview
This PR includes several changes in the parsing logic of the editor, specifically:
- The calculation and annotation of properties/backlinks has been moved out of the `say-parser` logic/module into its own `rdfa-annotator` module.
- The calculation and annotation of properties/backlinks is no longer executed in each `parse` call of the `say-parser` prosemirror parser in order to drastically improve performance. It is now only executed in 2 specific cases:
  * When setting the content of the editor using `setHTMLContent` or `initialize`
  * When pasting html content from the clipboard

When pasting html content from the clipboard, the following steps are executed:
- The HTML content of the clipboard is annotated (calculation of properties/backlinks)
- The HTML content of the clipboard is pasted
- The document is reloaded (in order to ensure the properties/backlinks are correctly calculated)

##### connected issues and PRs:
[GN-4579](https://binnenland.atlassian.net/browse/GN-4579?atlOrigin=eyJpIjoiNTRmZTI1ZmE0ZWNjNDIxZmFkNzIyMDFkNTY0MTIyZWQiLCJwIjoiaiJ9)

### How to test/reproduce
- Set the content of the document using one of the sample-data configurations
- Ensure that the backlinks and properties are correctly calculated (e.g. for the example decision or rdfa-test)
- Copy and paste the content of the document (or other rdfa) and check if the backlinks/properties are correctly calculated for the pasted content

### Challenges/uncertainties
- I am not sure about the naming of the new functions yet, but could not yet find a better name than `annotate`.
- Unsure where we should define the `Property`/`Backlink` types

### Checks PR readiness
